dynamiqs/sesolve/sesolve.py

```diff
@@ -106,9 +111,20 @@ def _vmap_sesolve(
 ) -> SEResult:
     # === vectorize function
     # we vectorize over H and psi0, all other arguments are not vectorized
-    is_batched = (H.ndim > 2, psi0.ndim > 2, False, False, False, False, False)
+    is_batched = (
+        is_timearray_batched(H),
```

Comment:
If we choose to do this we need to fix the type of `is_batched: list[bool | list[bool]]` in `compute_vmap`. But let's maybe wait to see if there's another solution.

Reply:
What's wrong with this solution? :D

Reply:
Mmmm it's just that how things are batched becomes quite hard to understand, we're in between supporting lists or supporting full PyTrees of H/Ls/rhos, we're "kind of supporting more advanced batching than just 1d per argument but with this, that and this constraint". I think the two extremes (1d list only, full PyTrees) are easier to write, easier to read, and easier to explain. Cf https://linear.app/dynamiqs/issue/DYN-157/support-batching-on-pytrees. That's why I'm not 100% satisfied with the solution, but I agree it's good enough for now. 👍